llama_index/indices/vector_store/auto_retriever/auto_retriever.py

```diff
@@ -0,0 +1,73 @@
+
+
+
+import logging
+from typing import Any, Dict, List, cast
+
+from llama_index.data_structs.node import NodeWithScore
+from llama_index.indices import base_retriever
+from llama_index.indices.query.schema import QueryBundle
+from llama_index.indices.vector_store.auto_retriever.output_parser import \
+    VectorStoreQueryOutputParser
+from llama_index.indices.vector_store.auto_retriever.prompts import (
+    DEFAULT_VECTOR_STORE_QUERY_PROMPT_TMPL, VectorStoreQueryPrompt)
+from llama_index.indices.vector_store.auto_retriever.schema import \
+    VectorStoreInfo
+from llama_index.indices.vector_store.base import GPTVectorStoreIndex
+from llama_index.indices.vector_store.retrievers import VectorIndexRetriever
+from llama_index.output_parsers.base import StructuredOutput
+from llama_index.vector_stores.types import (MetadataFilters,
+                                             VectorStoreQuerySpec)
+
+_logger = logging.getLogger(__name__)
+
+class VectorIndexAutoRetriever(base_retriever.BaseRetriever):
+    def __init__(
+            self,
+            index: GPTVectorStoreIndex,
+            vector_store_info: VectorStoreInfo,
+            **kwargs: Any,
+        ) -> None:
+        self._index = index
+        self._vector_store = self._index.vector_store
+        self._service_context = self._index.service_context
+        self._docstore = self._index.docstore
+
+        self._kwargs: Dict[str, Any] = kwargs.get("retriever_kwargs", {})
```

Comment:
this could just be a kwarg in the `__init__ `

Reply:
removed